### PR TITLE
lua: Test byte_extract access from match scripts

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #591 

Add tests for accessing byte_extract and byte_math variables from lua match scripts

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [#2871](https://redmine.openinfosecfoundation.org/issues/2871)

Updates:
- Remove use of Lua 5.3 `string.pack` in tests
- Validate values with hex comparisons

### CODE REVIEW

**Comment by catenacyber** [comment]
Suricata PR was merged in https://github.com/OISF/suricata/pull/7663

**Review by catenacyber — CHANGES_REQUESTED** [review]
Make min-version 7 so that all tests pass

File: tests/lua-byte-extract/test.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+requires:
+  features:
+    - HAVE_LUA
```
[the remark this answers is not in the document]
Comment (jlucovsky) [reply tests/lua-byte-extract/test.yaml:5]: Added `min-version: 7`

**Comment by victorjulien** [comment]
Merged in #932, thanks!

**Pull request closed without merging**